storage/innobase/lock/lock0lock.cc

```diff
@@ -584,141 +597,177 @@ ATTRIBUTE_NOINLINE static bool wsrep_is_BF_lock_timeout(const trx_t &trx)
 }
 #endif /* WITH_WSREP */
 
-/*********************************************************************//**
-Checks if a lock request for a new lock has to wait for request lock2.
+/** Checks if a lock request for a new lock has to wait for request lock2.
+@param trx                  trx of new lock
+@param type_mode            precise mode of the new lock
+                            to set: LOCK_S or LOCK_X, possibly
+                            ORed to LOCK_GAP or LOCK_REC_NOT_GAP,
+                            LOCK_INSERT_INTENTION.
+@param lock2                another record lock; NOTE that
+                            it is assumed that this has a lock bit
+                            set on the same record as in the new
+	                    lock we are setting.
+@param lock_is_on_supremum  TRUE if we are setting the
+                            lock on the 'supremum' record of an
+                            index page: we know then that the lock
+	                    request is really for a 'gap' type lock.
 @return TRUE if new lock has to wait for lock2 to be removed */
-UNIV_INLINE
-bool
-lock_rec_has_to_wait(
-/*=================*/
-	const trx_t*	trx,	/*!< in: trx of new lock */
-	unsigned	type_mode,/*!< in: precise mode of the new lock
-				to set: LOCK_S or LOCK_X, possibly
-				ORed to LOCK_GAP or LOCK_REC_NOT_GAP,
-				LOCK_INSERT_INTENTION */
-	const lock_t*	lock2,	/*!< in: another record lock; NOTE that
-				it is assumed that this has a lock bit
-				set on the same record as in the new
-				lock we are setting */
-	bool		lock_is_on_supremum)
-				/*!< in: TRUE if we are setting the
-				lock on the 'supremum' record of an
-				index page: we know then that the lock
-				request is really for a 'gap' type lock */
-{
-	ut_ad(trx);
-	ut_ad(!lock2->is_table());
-	ut_d(lock_sys.hash_get(type_mode).assert_locked(
-		     lock2->un_member.rec_lock.page_id));
+inline
+bool lock_rec_has_to_wait(const trx_t *trx, unsigned type_mode,
+                          const lock_t* lock2, bool lock_is_on_supremum)
+{
+  ut_ad(trx);
+  ut_ad(!lock2->is_table());
+  ut_d(lock_sys.hash_get(type_mode).assert_locked(
+         lock2->un_member.rec_lock.page_id));
+
+  if (trx == lock2->trx
+      || lock_mode_compatible(
```

Comment:
If you are reformatting a function to follow the common formatting rules, then you should consistently follow those rules:
* Never break a line before any binary operator such as `||`.
* Never break a line after `(`.